### PR TITLE
Initial TOC draft for the language specification.

### PR DESCRIPTION
**Pull request by sylvanc**

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- `_` as "don't care".
+
+### Strings
+
+### Constants
+
+- Boolean
+- Various integers and floats
+
+> How to talk about integer literals not having a type yet?
```
Comment (sophiaIC) [inline spec/spec.md:18]: no strings?

File: spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+- `_` as "don't care".
+
+### Strings
+
+### Constants
+
+- Boolean
+- Various integers and floats
+
```
Comment (sophiaIC) [inline spec/spec.md:17]: no string constants?

Comment (sophiaIC) [inline spec/spec.md:17]: are strings not constant?

File: spec/spec.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+> How to talk about integer literals not having a type yet?
+
+### Lambdas
+
+### Object Literals
+
+- Open issues: code reuse, possible conflict with "real" `new`.
+
+### Expression Termination
+
```
Comment (sophiaIC) [inline spec/spec.md:27]: syntactic termination?

File: spec/spec.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+### Dots
+
+- Selectors
+- Reverse application
+
+### Universal Call Syntax
+
+- Object-like vs. operator-like local names
+  - Type names are `create` sugar
```
Comment (sophiaIC) [inline spec/spec.md:69]: do no issues arise about order of intiailization of objects?

File: spec/spec.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+### Generics and Predicates
+
+### Type Lists
+
+- Generics for tuple arity.
+
+### Capabilities
+
+> Chicken and egg with regions.
```
Comment (mjp41) [inline spec/spec.md:129]: I think we should discuss in the Teams call.

File: spec/spec.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+- Boolean
+- Various integers and floats
+
+> How to talk about integer literals not having a type yet?
+
+### Lambdas
+
+### Object Literals
+
+- Open issues: code reuse, possible conflict with "real" `new`.
```
Comment (TobiasWrigstad) [inline spec/spec.md:24]: ?

File: spec/spec.md (hunk 1 above)
Comment (mjp41) [inline spec/spec.md:18]: By this we mean `0` is the zero that is in U64, float, etc.

File: spec/spec.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+### Try
+
+- "Give me the value instead of returning"
+
+### Pattern Matching
+
+- Do we need to talk about type lists here?
+
+## Types
+
```
Comment (TobiasWrigstad) [inline spec/spec.md:104]: Where under here (or elsewhere) do the Store[] types go?

File: spec/spec.md (hunk 1 above)
Comment (nwf-msr) [inline spec/spec.md:18]: Someone somewhere surely has said something useful about Haskell's handling of (initially numeric, and now also string and list) literals.  https://www.haskell.org/tutorial/numbers.html is surely someone and somewhere, but as to something useful, I'll leave judgement to the reader.

File: spec/spec.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+- Fields
+- Functions
+- "Left-hand" functions
+- Nesting
+- Default values
+- Visibility and encapsulation
+- Generics: what needs to be discussed here as opposed to in `Types`?
+  - Treat it as template-like, discuss type checking later
+- Do we need to talk about type lists here?
+
```
Comment (TobiasWrigstad) [inline spec/spec.md:45]: What is the exception story?

File: spec/spec.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+### Comments
+
+### Identifiers and Symbols
+
+- `_` as "don't care".
+
+### Strings
+
+### Constants
```
Comment (mjp41) [inline spec/spec.md:13]: @davidchisnall Rename to "Primitive literal", rather than constants

File: spec/spec.md (hunk 2 above)
Comment (mjp41) [reply spec/spec.md:17]: Perhaps call the section "String literals" and move after "object literals"

Comment (mjp41) [reply spec/spec.md:17]: Do we have "array literals"?

File: spec/spec.md (hunk 6 above)
Comment (mjp41) [reply spec/spec.md:24]: Currently the parser uses `new` for string literals, but that might not be the final choice, and we should decide.

File: spec/spec.md (hunk 4 above)
Comment (mjp41) [reply spec/spec.md:69]: Comments from discussion
* Static initialisers in other languages, and when things happen. 
* Immutable static data initialisation, probably not a problem here.
* Initialisation order is complex in some languages.
* `create` as a factory method, and allocates after it has found all the bits. So create can fail, make sure we allow failure returns from `create`.

File: spec/spec.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
+# Verona Language Spec
+
```
Comment (mjp41) [inline spec/spec.md:2]: High-level comment.

Rationale subsections, or a separate rationale document.

File: spec/spec.md (hunk 7 above)
Comment (mjp41) [reply spec/spec.md:104]: In the assignment section above?

File: spec/spec.md (hunk 5 above)
Comment (mjp41) [reply spec/spec.md:129]: Comments:
* Describe the region model before the types to capture and enforce it.  
* Object model section before types - pretty pictures
* Types turn pretty pictures into maths.

**Pull request merged**